### DISCUSSION TITLE
Issue using x-cloak on an accordion

### DISCUSSION THREAD

**Discussion by benoitbohnke** [alpinejs/alpine#4074] [1. Help] [answered]
I'm having an issue regarding my nav menu that is a fixed div overlay on top of everything. My menu contains a series of accordion that  needs to be opened/closed depending on the page status. As the documentation says, I'm also using x-cloak to prevent my overlay to “blip” on page loads. 
Problem is it messing with my accordion logic which using max-height to show/hide content. As I'm using x-cloak, I think there is an issue calculating accordion height.
I understand this is not really an Alpine.js problem but Is someone could help me with that issue ?
I've build up a demo to show the problem (first accordion should be open when opening menu)
https://codepen.io/mi-re/pen/RwOPyRE

**Comment by ekwoka** [comment, marked as answer]
Is there a reason you don't want to use x-collapse?

I believe it will handle your issue for you.

But yes, elements not displayed in the layout don't have scrollHeight

[Implemented with x-collapse For reference](https://awesomealpine.com/play?coreplugins=2&script=Y29uc29sZS5sb2coJ2hlbGxvJyk7Cg&html=PGRpdiB4LWRhdGE9InttZW51T3BlbjogZmFsc2V9Ij4KICA8YnV0dG9uCiAgICBjbGFzcz0iYmctenluYy0yMDAiCiAgICBAY2xpY2s9Im1lbnVPcGVuID0gISBtZW51T3BlbiIKICA-CiAgICBTaG93IE1lbnUKICA8L2J1dHRvbj4KICA8ZGl2CiAgICBjbGFzcz0iYmctZ3JheS0xMDAgaC1zY3JlZW4gdy1zY3JlZW4gZmxleCBqdXN0aWZ5LWNlbnRlciIKICAgIHgtc2hvdz0ibWVudU9wZW4iCiAgICBzdHlsZT0iZGlzcGxheTogbm9uZTsiCiAgPgogICAgPHVsPgogICAgICA8bGkgeC1kYXRhPSJ7b3Blbjp0cnVlfSI-CiAgICAgICAgPGEKICAgICAgICAgIEBjbGljay5wcmV2ZW50PSJvcGVuID0gISBvcGVuIgogICAgICAgICAgaHJlZj0iIyIKICAgICAgICA-CiAgICAgICAgICBJdGVtIDEKICAgICAgICA8L2E-CiAgICAgICAgPHVsCiAgICAgICAgICBjbGFzcz0idHJhbnNpdGlvbi1hbGwgZHVyYXRpb24tMzAwIgogICAgICAgICAgeC1yZWY9ImNvbnRlbnQiCiAgICAgICAgICB4LXNob3c9Im9wZW4iCiAgICAgICAgICB4LWNvbGxhcHNlCiAgICAgICAgPgogICAgICAgICAgPGxpPlN1Yml0ZW0gMTwvbGk-CiAgICAgICAgICA8bGk-U3ViaXRlbSAyPC9saT4KICAgICAgICAgIDxsaT5TdWJpdGVtIDM8L2xpPgogICAgICAgICAgPGxpPlN1Yml0ZW0gNDwvbGk-CiAgICAgICAgPC91bD4KICAgICAgPC9saT4KICAgICAgPGxpIHgtZGF0YT0ie29wZW46ZmFsc2V9Ij4KICAgICAgICA8YQogICAgICAgICAgQGNsaWNrLnByZXZlbnQ9Im9wZW4gPSAhIG9wZW4iCiAgICAgICAgICBocmVmPSIjIgogICAgICAgID4KICAgICAgICAgIEl0ZW0gMgogICAgICAgIDwvYT4KICAgICAgICA8dWwKICAgICAgICAgIGNsYXNzPSJ0cmFuc2l0aW9uLWFsbCBkdXJhdGlvbi0zMDAiCiAgICAgICAgICB4LXJlZj0iY29udGVudCIKICAgICAgICAgIHgtc2hvdz0ib3BlbiIKICAgICAgICAgIHgtY29sbGFwc2UKICAgICAgICA-CiAgICAgICAgICA8bGk-U3ViaXRlbSAxPC9saT4KICAgICAgICAgIDxsaT5TdWJpdGVtIDI8L2xpPgogICAgICAgICAgPGxpPlN1Yml0ZW0gMzwvbGk-CiAgICAgICAgICA8bGk-U3ViaXRlbSA0PC9saT4KICAgICAgICA8L3VsPgogICAgICA8L2xpPgogICAgICA8bGkgeC1kYXRhPSJ7b3BlbjpmYWxzZX0iPgogICAgICAgIDxhCiAgICAgICAgICBAY2xpY2sucHJldmVudD0ib3BlbiA9ICEgb3BlbiIKICAgICAgICAgIGhyZWY9IiMiCiAgICAgICAgPgogICAgICAgICAgSXRlbSAzCiAgICAgICAgPC9hPgogICAgICAgIDx1bAogICAgICAgICAgY2xhc3M9InRyYW5zaXRpb24tYWxsIGR1cmF0aW9uLTMwMCIKICAgICAgICAgIHgtcmVmPSJjb250ZW50IgogICAgICAgICAgeC1zaG93PSJvcGVuIgogICAgICAgICAgeC1jb2xsYXBzZQogICAgICAgID4KICAgICAgICAgIDxsaT5TdWJpdGVtIDE8L2xpPgogICAgICAgICAgPGxpPlN1Yml0ZW0gMjwvbGk-CiAgICAgICAgICA8bGk-U3ViaXRlbSAzPC9saT4KICAgICAgICAgIDxsaT5TdWJpdGVtIDQ8L2xpPgogICAgICAgIDwvdWw-CiAgICAgIDwvbGk-CiAgICA8L3VsPgogIDwvZGl2Pgo8L2Rpdj4K&tw=1)

**Reply by benoitbohnke** [reply]
Thanks I wasn't aware of x-collapse and it solves my problem. 
(sorry for the confusing pen..)
